tools/logstash/lib/main.sc

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env amm
+
+import $file.dependencies
+import $file.util
+import $file.logformat
+
+import logformat.LogFormat
+import scala.annotation.tailrec
+import akka.actor.ActorSystem
+import akka.stream.{ ActorMaterializer, Materializer }
+import akka.stream.IOResult
+import akka.util.ByteString
+import akka.stream.scaladsl._
+import ammonite.ops._
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.{Try, Success, Failure}
+
+@tailrec def await[T](f: Future[T]): T = f.value match {
+  case None =>
+    Thread.sleep(10)
+    await(f)
+  case Some(v) =>
+    v.get
+}
```

Comment:
Interesting. I've done it with

```scala
val done = Promise[Done]
done.completeWith(f)
while (!done.isCompleted) { Thread.sleep(100) }
```

probably boils down to the same thing 🙂.

Reply:
yep :)